hedera-node/hedera-admin-service-impl/src/main/java/com/hedera/node/app/service/admin/impl/handlers/UnzipUtility.java

```diff
@@ -0,0 +1,108 @@
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Utility class to unzip a zip file.
+ * <p>
+ * This class was copied from hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/utils/UnzipUtility.java
+ * Once we are wholly migrated to the modularized services, we can remove the other location.
+ * */
+public final class UnzipUtility {
+    private static final Logger log = LogManager.getLogger(UnzipUtility.class);
+
+    private static final int BUFFER_SIZE = 4096;
+
+    private UnzipUtility() {
+        throw new UnsupportedOperationException("Utility Class");
+    }
+
+    public static boolean unzip(final byte[] bytes, final String dstDir) throws IOException {
+        final File destDir = new File(dstDir);
+        if (!destDir.exists()) {
+            if (!destDir.mkdirs()) {
+                log.fatal("Unable to create the directory for update assets: {}", destDir);
+                return false;
+            }
+            log.info("Created directory {} for update assets", destDir);
+        }
+
+        final var zipIn = new ZipInputStream(new ByteArrayInputStream(bytes));
+        ZipEntry entry = zipIn.getNextEntry();
+        if (entry == null) return false;
+        while (entry != null) {
+            var filePath = dstDir + File.separator + entry.getName();
+            final var fileOrDir = new File(filePath);
+            filePath = fileOrDir.getCanonicalPath();
+
+            if (!filePath.startsWith(destDir.getCanonicalPath() + File.separator)) {
+                throw new IllegalArgumentException("Zip file entry " + filePath + " has an invalid path prefix!");
+            }
+
+            if (!entry.isDirectory()) {
+                ensureDirectoriesExist(fileOrDir);
+                extractSingleFile(zipIn, filePath);
+                log.info(" - Extracted update file {}", filePath);
+            } else {
+                if (!fileOrDir.mkdirs()) {
+                    log.error(" - Unable to create assets sub-directory: {}", fileOrDir);
+                }
+                log.info(" - Created assets sub-directory {}", fileOrDir);
+            }
+            zipIn.closeEntry();
+            entry = zipIn.getNextEntry();
+        }
+        zipIn.close();
+        return true;
+    }
+
+    /**
+     * Extracts a zip entry (file entry)
+     *
+     * @param inputStream Input stream of zip file content
+     * @param filePath Output file name
+     */
+    public static void extractSingleFile(ZipInputStream inputStream, String filePath) {
+        try (var bos = new BufferedOutputStream(new FileOutputStream(filePath))) {
+            final var bytesIn = new byte[BUFFER_SIZE];
+            int read;
+            while ((read = inputStream.read(bytesIn)) != -1) {
+                bos.write(bytesIn, 0, read);
+            }
+        } catch (IOException e) {
+            log.error("Unable to write to file {}", filePath, e);
+        }
+    }
+
+    static void ensureDirectoriesExist(final File file) {
```

Comment:
I do not understand the use case of this method

Reply:
It's not strictly necessary for it to be a separate method. Since it's only used in one place it could be folded into the code where it's called. However, IMO it adds to readability, since 'ensureDirectoriesExist' describes what it does: to ensure that the directory of the passed in file path exists, creating the directory if necessary.